units/patch_intent.go

```diff
@@ -527,30 +526,38 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 		if err = setTasksToPreviousFailed(patchDoc, previousPatch, project); err != nil {
 			return "", errors.Wrap(err, "settings tasks to previous failed")
 		}
-
 	} else {
 		patchDoc.Tasks = previousPatch.Tasks
 	}
 
 	return previousPatch.Status, nil
 }
 
-func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, error) {
+func getPreviousFailedTasksAndDisplayTasks(project *model.Project, vt patch.VariantTasks, version string) ([]string, error) {
+	tasksInProjectVariant := project.FindTasksForVariant(vt.Variant)
 	failedTasks, err := task.FindAll(db.Query(task.FailedTasksByVersionAndBV(version, vt.Variant)))
 	if err != nil {
 		return nil, errors.Wrapf(err, "finding failed tasks in build variant '%s' from previous patch '%s'", vt.Variant, version)
 	}
-	failedExecutionTasks := []string{}
+	// Verify that the taskgroup/task is in the current project definition and in the previous run.
+	allFailedTasks := []string{}
 	for _, failedTask := range failedTasks {
-		if !failedTask.DisplayOnly {
-			failedExecutionTasks = append(failedExecutionTasks, failedTask.DisplayName)
+		if failedTask.TaskGroup != "" &&
+			utility.StringSliceContains(tasksInProjectVariant, failedTask.TaskGroup) &&
+			utility.StringSliceContains(vt.Tasks, failedTask.DisplayName) {
```

Comment:
Can we check this above and just "continue" since we check it in both the if and else case? And then we can add a comment like "skip failed tasks that aren't in the patch definition"  (Although is that a new thing? I don't really understand why we need it; theoretically anything returned from the failed tasks query is part of the patch definition or it wouldn't be a failed task)

Reply:
Done, I wrapped it in an extra if 

I'm also not super sure why it's necessary to verify this at this point 
I think @hadjri worked on this last so I wanted to ask if this was necessary for a particular use case?